### PR TITLE
Add extension replacing

### PR DESCRIPTION
**Pull request by Banou26**
*fix https://github.com/karlprieb/babel-plugin-add-import-extension/issues/6*

In the end what you proposed was enough for me so i didn't implement the regex replacer.
I also did some refactoring of the code to simplify it, since we had a lot of duplicated code.

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -15,76 +15,64 @@ const isNodeModule = module => {
...
-const skipModule = module => isNodeModule(module) || extname(module).length > 0
+const makeDeclaration =
+  ({ declaration, args, replace = false, extension = 'js' }) =>
+      (path, { file: { opts: { filename } }}) => {
+        const { node: { source } } = path
+
+        if (!source || skipModule(source && source.value, { replace, extension })) return
+        const { value: module } = source
+
+        if (module[0] === '.') {
```
Comment (Banou26) [inline src/plugin.js]: We could actually move that test to skipModule

File: src/plugin.js [hunk 2]
```diff
@@ -15,76 +15,64 @@ const isNodeModule = module => {
...
+const skipModule = (module, { replace, extension }) =>
+  isNodeModule(module)
+  || (
+    replace
+      ? extname(module) === `.${extension}`
+      : extname(module).length
+  )
 
-const skipModule = module => isNodeModule(module) || extname(module).length > 0
+const makeDeclaration =
```
Comment (karlprieb) [inline src/plugin.js]: Great job :D

File: src/plugin.js (hunk 1 above)
Comment (karlprieb) [reply src/plugin.js]: I agreed, can you do that?

File: tests/test.js [hunk 3]
```diff
@@ -22,9 +22,9 @@ export { transform } from '@babel/core'
 describe('Replace', () => {
   test.each`
     type                             | statements          | extension
-    ${'default extension to import'} | ${importStatements} | ${false}
+    ${'default extension to import'} | ${importStatements} | ${undefined}
```
Comment (karlprieb) [inline tests/test.js]: 👍

File: src/plugin.js [hunk 4]
```diff
@@ -15,76 +15,64 @@ const isNodeModule = module => {
...
+  ({ declaration, args, replace = false, extension = 'js' }) =>
```
Comment (karlprieb) [inline src/plugin.js]: From where did you get the `replace` arg?

Comment (Banou26) [reply src/plugin.js]: Yeah, the object i'm destructuring is a merge of this [object](https://github.com/karlprieb/babel-plugin-add-import-extension/pull/7/files/3bbe6f988abf49a89878c8bfa452f88dcb74f3bc#diff-a4f485aa834ca5b3c24819fceb39c56fR61-R65), and the `replace` is from coming from the `option` object, itself coming from the `.babelrc` options for `babel-plugin-add-import-extension`

Comment (karlprieb) [reply src/plugin.js]: 👍

File: tests/test.js (hunk 3 above)
Comment (karlprieb) [reply tests/test.js]: I think we can change this "table" to add the `replace` option. What you think?

File: src/plugin.js (hunk 1 above)
Comment (Banou26) [reply src/plugin.js]: Sure

**Comment by karlprieb** [comment]
Can you add the example with the `replace` option on README?

File: tests/test.js (hunk 3 above)
Comment (Banou26) [reply tests/test.js]: Yeah, i forgot to add tests for the `replace` option

Comment (Banou26) [reply tests/test.js]: Done

File: src/plugin.js (hunk 1 above)
Comment (Banou26) [reply src/plugin.js]: Done

File: tests/test.js [hunk 5]
```diff
@@ -17,18 +24,27 @@ export { something as another } from './lib/something'
...
-    ${'default extension to export'} | ${exportStatements} | ${false}
-    ${'custom extension to export'}  | ${exportStatements} | ${'jsx'}
-  `('should add the $type statements', ({statements, extension}) => {
+    type                             | statements          | extension    | replace
+    ${'default extension to import'} | ${importStatements} | ${undefined} | ${undefined}
+    ${'custom extension to import'}  | ${importStatements} | ${'jsx'}     | ${undefined}
+    ${'default extension to export'} | ${exportStatements} | ${undefined} | ${undefined}
+    ${'custom extension to export'}  | ${exportStatements} | ${'jsx'}     | ${undefined}
+    ${'replace extension to import'} | ${importStatements} | ${undefined} | ${true}
+    ${'replace extension to import'} | ${importStatements} | ${'js'}     | ${true}
```
Comment (karlprieb) [inline tests/test.js]: maybe we can use another extension here so we can make sure that works, because `js` is the default extension. What you think?

**Comment by karlprieb** [comment]
Great PR and changes @Banou26 :D Just one concern on the test file.

File: tests/test.js (hunk 5 above)
Comment (Banou26) [reply tests/test.js]: yup

Comment (Banou26) [reply tests/test.js]: Done

**Comment by karlprieb** [comment]
Thank you! I released the 1.3.0 version with this feature!

**Pull request merged**